webrender/src/device.rs

```diff
@@ -1391,27 +1387,28 @@ impl Device {
         let vs_id = Device::compile_shader(&program.name,
                                            &program.vs_source,
                                            gl::VERTEX_SHADER,
-                                           &vs_preamble,
-                                           panic_on_fail);
+                                           &vs_preamble);
         let fs_id = Device::compile_shader(&program.name,
                                            &program.fs_source,
                                            gl::FRAGMENT_SHADER,
-                                           &fs_preamble,
-                                           panic_on_fail);
+                                           &fs_preamble);
 
         match (vs_id, fs_id) {
-            (Some(vs_id), None) => {
+            (Ok(vs_id), Err(e)) => {
                 println!("FAILED to load fs - falling back to previous!");
```

Comment:
I don't think failing back to a previous shader is a good idea, especially if we get in a partial state where, for example, the FS is new and the VS is old.
It would be much nicer to do this idiomatically, as in - `let vs_id = try!(Device::compile_shader(..));` without the need to do this match afterwards.